### ISSUE TITLE
gphoto2 capture image and download fails after multiple captures in RAW format

### ISSUE THREAD

**Issue by joelcomp1** [gphoto/gphoto2#176]
I currently use a Canon Mark III camera for automated captures using the RAW image format.  This works fine most of the time but sometimes after a few captures it will start to fail with:

ERROR: Could not capture image.  ERROR: Could not capture.  

I am using --capture-image-and-download to take the image and download to my machine.  When it gets into this state I can only resolve by restarting the camera on and off and then it will be fine again for a while.  I also noticed when in this state if I change the format to say Fine JPEG it captures fine but if I use any RAW format it appears to timeout.  I am currently using libgphoto2 2.5.19 and gphoto2 2.5.17.  I quickly tried to use 2.5.20 (latest version) but saw same result.

**Comment by joelcomp1** [comment]
I just noticed that when this happens my availableshots is at zero (I am using RAM only no SDCARD) what does available shots represent?  It seems when I reset the camera this goes back to 11188

**Comment by msmeissn** [comment]
what is capturetarget being set? If it is RAM, you need to download all the images. (so no --keep-raw)

Our code might have bugs and if it misses to download images, it would fill up this RAM.

**Comment by msmeissn** [comment]
hmm, how many images are you able to capture and what is their approximate size?

**Comment by msmeissn** [comment]
so you are capturing a lot of images? so thousands?

**Comment by joelcomp1** [comment]
yup but over weeks of time we just don't turn the cameras off, we found the current work around is to just power cycle the camera before we get to 0.   I guess that count is indicative of some memory left on the camera?

**Comment by msmeissn** [comment]
actually its a curios thing. the computer sends to the EOS how much Harddisk space it has so the EOS knows how many pictures the computer would fit.

As we currently only send this info once on startup, several thousand RAWs will drain it over time.

I will add code that does refresh this more often.

**Comment by msmeissn** [comment]
(so a workaround should be to restart gphoto2 I think.)

**Comment by joelcomp1** [comment]
Hmm interesting so your saying its reading the HDD on my PC and using that to compute the available shots? We re-start gphoto2 for each capture so I don't think that is it.  The only thing that isn't restarting is the camera itself we will leave it on for days/weeks.

**Comment by msmeissn** [comment]
the camera driver sends over to the camera. libgphoto2 currently used a value around of 5GB, which is a bit small and would run full in your case of thousands of RAW images.

commit 8d82d383b9a0603fbe9f374403434dbd1f943b73  as test, where we reset the available size before every capture.

workaround might not work, as we optimized it away potentially

**Comment by msmeissn** [comment]
(commit is in libgphoto2 git )

**Comment by joelcomp1** [comment]
awesome thanks! Trying out now and will let you know

**Comment by joelcomp1** [comment]
Tried it out in our setup if I reboot the camera I am getting availableshots equal to 35808 now instead of the 11188.  Is that computing based on HDD size or just fixed? seems fixed from your commit, and I tried "filling up" my HDD but didn't seem to affect that output number.

Only odd thing was (not sure your intent) but if I take a picture the number still decrements roughly for each one 35807, 35806 etc but then if I power cycle its back to 35808.  I noticed in your commit a4a510fd60f177c3a7e4b239ad76298bc2502615 you only reset the HDD size if available shots less then 100?  So that maybe does work just haven't taken nearly enough images.

**Comment by msmeissn** [comment]
yes, the observation is correct.

- i bumped the "fake" size reported, so the *3 jump is apparent
- i added code if it goes below 100, although I also did not test it
